gloo/transport/uv/context.h

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) 2019-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <deque>
+#include <memory>
+#include <mutex>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+
+#include <gloo/common/memory.h>
+#include <gloo/transport/context.h>
+
+//
+// READ THIS FIRST!
+//
+// This file is a copy of "gloo/transport/tcp/context.h".
+//
+// Any modifications should be made to that file as well. This file is
+// not a top level construct because it assumes the transport supports
```

Comment:
Should we then extract these to some `UnboundedBufferContext` and put that as a top level abstract class? Don't have to be done in this PR.

Reply:
Yes, it can and should be pulled to the top level since there is nothing transport specific in it.

Reply:
I created #207 to track.